prow/plugins/hold/hold.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package hold contains a plugin which will allow users to label their
+// own pull requests as not ready or ready for merge. The submit queue
+// will honor the label to ensure pull requests do not merge when it is
+// applied.
+package hold
+
+import (
+	"regexp"
+
+	"github.com/Sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const pluginName = "hold"
+
+var (
+	label         = "do-not-merge/hold"
+	labelRe       = regexp.MustCompile(`(?mi)^/hold\s*$`)
+	labelCancelRe = regexp.MustCompile(`(?mi)^/hold cancel\s*$`)
+)
+
+type event struct {
+	org           string
+	repo          string
+	number        int
+	prAuthor      string
+	commentAuthor string
+	body          string
+	htmlurl       string
+	hasLabel      bool
+}
+
+func init() {
+	plugins.RegisterIssueCommentHandler(pluginName, handleIssueComment)
```

Comment:
This should also work with a review comment, in addition to an issue comment

See https://github.com/kubernetes/test-infra/blob/master/prow/plugins/lgtm/lgtm.go#L52

Reply:
Good point! I'll add that.

Reply:
I'll also add support for it being in the review body since we do that in `/lgtm`